# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/WeakMap.h
# Commit: 5a6684922b98
# Full Hash: 5a6684922b980d32b05bd7210cfadc34e734a845
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-03-21 21:28:09
# Regressor Bug: 1623973
# File Overlap Count: 2
# Description:
#   Bug 1623973 - Make FinalizationRegistryObject active record set and registrations map values weak r=sfink
#   
#   This removes tracing for the values of the registration map and the active record set and sweeps them instead. This requires maintaining a map of FinalizationRegistryObjects on the zone.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D67677
# ==============================================================================

diff -r c4d2ca8f78b7 -r 5a6684922b98 js/src/gc/WeakMap.h
--- a/js/src/gc/WeakMap.h	Thu Mar 19 16:42:42 2020 +0000
+++ b/js/src/gc/WeakMap.h	Fri Mar 20 21:57:22 2020 +0000
@@ -313,6 +313,8 @@
     return mallocSizeOf(this) + sizeOfExcludingThis(mallocSizeOf);
   }
 
+  ObjectValueWeakMap& valueMap() { return map; }
+
 #ifdef JSGC_HASH_TABLE_CHECKS
   void checkAfterMovingGC();
 #endif